src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -148,146 +149,396 @@ final case class ThermalGrid(
           val (fullHouseState, maybeFullHouseThreshold) =
             thermalHouse.determineState(
               relevantData,
               lastHouseState,
               lastAmbientTemperature,
               zeroKW,
             )
-          storage.zip(updatedStorageState) match {
-            case Some((thermalStorage, storageState)) =>
-              val (updatedStorageState, maybeStorageThreshold) =
-                thermalStorage.updateState(
-                  relevantData.currentTick,
-                  qDot,
-                  storageState,
-                )
-
-              /* Both house and storage are updated. Determine what reaches the next threshold */
-              val nextThreshold = determineMostRecentThreshold(
-                maybeFullHouseThreshold,
-                maybeStorageThreshold,
-              )
-
-              (
-                state.copy(
-                  houseState = Some(fullHouseState),
-                  storageState = Some(updatedStorageState),
-                ),
-                nextThreshold,
-              )
-            case None =>
-              /* There is no storage, house determines the next activation */
-              (
-                state.copy(houseState = Some(fullHouseState)),
-                maybeFullHouseThreshold,
-              )
-          }
+          (Some(fullHouseState), maybeFullHouseThreshold, qDotHouse)
         } else {
-          /* The house can handle the infeed */
-          (
-            state.copy(houseState = Some(updatedHouseState)),
-            maybeHouseThreshold,
-          )
+          (Some(newState), threshold, zeroKW)
         }
+      case _ => (None, None, zeroKW)
+    }
+  }
 
-      case None =>
-        storage.zip(state.storageState) match {
-          case Some((thermalStorage, storageState)) =>
-            val (updatedStorageState, maybeStorageThreshold) =
-              thermalStorage.updateState(
-                relevantData.currentTick,
-                qDot,
-                storageState,
-              )
-            (
-              state.copy(storageState = Some(updatedStorageState)),
-              maybeStorageThreshold,
-            )
-          case None =>
-            throw new InconsistentStateException(
-              "A thermal grid has to contain either at least a house or a storage."
-            )
-        }
+  /** Handles the cases, when the storage has heat demand and will be filled up
+    * here (positive qDot) or will be return its stored energy into the thermal
+    * grid (negative qDot).
+    * @param tick
+    *   Current tick
+    * @param state
+    *   Current state of the houses
+    * @param qDotStorage
+    *   Infeed to the storage (positive: Storage is charging, negative: Storage
+    *   is discharging)
+    * @return
+    *   Updated thermal grid state
+    */
+  private def handleInfeedStorage(
```

Comment:
This is not only thermal feed-in, but also consumption, so maybe call this handleCasesStorage or similar?